### PR TITLE
Optimise `_checkProtocolFee`

### PR DESCRIPTION
**Pull request by malik672**
## Related Issue
Gas Optimisation

## Description of changes
1.Introduced unchecked block for the _checkProtocolFee() function , this potentially saves 100+ in gas
2.Performed modulo operation using bit operation this saves 25+ in gas, for more information check [http://graphics.stanford.edu/~seander/bithacks.html#ModulusDivisionEasy](https://github.com/Uniswap/v4-core/pull/here)
3.used uint256 as param type instead of uint8

### CODE REVIEW

File: contracts/PoolManager.sol [hunk 1]
```diff
@@ -433,15 +433,18 @@ contract PoolManager is IPoolManager, Owned, NoDelegateCall, ERC1155, IERC1155Re
...
-            uint8 fee1 = fee >> 4;
-            // The fee is specified as a denominator so it cannot be LESS than the MIN_PROTOCOL_FEE_DENOMINATOR (unless it is 0).
-            if (
-                (fee0 != 0 && fee0 < MIN_PROTOCOL_FEE_DENOMINATOR) || (fee1 != 0 && fee1 < MIN_PROTOCOL_FEE_DENOMINATOR)
-            ) {
-                revert FeeTooLarge();
+    function _checkProtocolFee(uint256 fee) internal pure {
+        unchecked {
+            if (fee > 0) {
+                uint256 fee0 = fee & (16 - 1);
```
Comment (hensha256) [inline contracts/PoolManager.sol:439]:
```suggestion
                uint256 fee0 = fee & 0xF;
```

**Comment by malik672** [comment]
will do that

**Comment by malik672** [comment]
@hensha256 done

**Comment by hensha256** [comment]
Hi @malik672 the snapshots are showing that there arent any gas improvements from your code changes? Only the bytecode size has changed

**Comment by malik672** [comment]
> Hi @malik672 the snapshots are showing that there arent any gas improvements from your code changes? Only the bytecode size has changed

probably because i re-cloned the repo i've already commit to then ran a yarn snapshots on it, could that be the reason?

**Comment by malik672** [comment]
> Hi @malik672 the snapshots are showing that there arent any gas improvements from your code changes? Only the bytecode size has changed

**Comment by hensha256** [comment]
We have now made a number of changes for how protocol fees work so this PR is out of date. We are closing this PR in favor of #283

**Pull request closed without merging**